pkg/watcher/reconciler/dynamic/dynamic_test.go

```diff
@@ -205,10 +206,33 @@ func TestReconcile_TaskRun(t *testing.T) {
 		}
 
 		_, err := trclient.Get(ctx, taskrun.GetName(), metav1.GetOptions{})
-		if !errors.IsNotFound(err) {
+		if !apierrors.IsNotFound(err) {
 			t.Fatalf("wanted NotFound, got %v", err)
 		}
 	})
+
+	t.Run("delete failed runs", func(t *testing.T) {
```

Comment:
I think this needs one more test case for "keep running TaskRuns" (the `Unknown` status.)